internal/runbits/legacy/projectmigration/projectmigration.go

```diff
@@ -78,16 +78,110 @@ func PromptAndMigrate(proj projecter) (bool, error) {
 		break
 	}
 
-	pf := projectfile.NewProjectField()
-	if err := pf.LoadProject(proj.URL()); err != nil {
-		return false, errs.Wrap(err, "Could not load activestate.yaml")
+	// Prevent also showing the warning when we already prompt
+	warned = true
+
+	defaultChoice := false
+	if migrate, err := prompter.Confirm("", locale.T("projectmigration_confirm"), &defaultChoice); err == nil && !migrate {
+		if out.Config().Interactive {
+			out.Notice(locale.T("projectmigration_declined"))
+		}
+		return CreateMigrateScript(proj)
+	} else if err != nil {
+		return errs.Wrap(err, "Could not confirm migration choice")
 	}
-	pf.StripCommitID()
-	if err := pf.Save(proj.Path()); err != nil {
-		return false, errs.Wrap(err, "Could not save activestate.yaml")
+
+	if err := proj.StripLegacyCommitID(); err != nil {
+		return errs.Wrap(err, "Could not strip legacy commit ID")
 	}
 
 	out.Notice(locale.Tl("projectmigration_success", "Your project was successfully migrated"))
 
-	return true, nil
+	return nil
+}
+
+var scriptsRx = regexp.MustCompile(`(?m)^scripts:\n`)
+
+func CreateMigrateScript(proj projecter) error {
+	scriptValue := migrateNixScript
+	scriptLanguage := "bash"
+	switch runtime.GOOS {
+	case "darwin":
+		scriptValue = migrateMacScript
+	case "windows":
+		scriptValue = migrateWinScript
+		scriptLanguage = "powershell"
+	}
+
+	script := projectfile.Script{
+		projectfile.NameVal{
+			Name:  "migrate-to-buildscripts",
+			Value: string(scriptValue),
+		},
+		projectfile.ScriptFields{
+			// No localized since this is a legacy migration mechanic and we'll remove this long before we'll introduce
+			// another language.
```

Comment:
I'm not following this comment. It looks like there is localization?

Reply:
Forgot to drop it after I moved it to locale cause it was getting noisy.